src/main/webapp/resources/js/pages/projects/reference-files/ReferenceFiles.jsx

```diff
@@ -0,0 +1,169 @@
+  }
+
+  // Columns for the reference file table
+  const referenceFileTableColumns = [
+    {
+      title: i18n("ReferenceFile.name"),
+      dataIndex: "label",
+    },
+    {
+      title: i18n("ReferenceFile.size"),
+      dataIndex: "size",
+    },
+    {
+      title: i18n("ReferenceFile.added"),
+      dataIndex: "createdDate",
+      render(text) {
+        return formatInternationalizedDateTime(text);
+      },
+    },
+    {
+      align: "right",
+      render(file) {
+        let actionButtons =
+          [
+            <DownloadTableItemButton
+              key={`download-btn-${file.id}`}
+              onDownload={() => downloadProjectReferenceFile(file.id)}
+              onDownloadSuccess={() => {
+                notification.success({message: i18n("ReferenceFile.downloadingFileSuccess", file.label, projectName)});
+              }}
+              tooltipText={i18n("ReferenceFile.downloadTooltip")}
+            />,
+            // Only display remove button for reference files if user can manage project
+            canManage ?
+              <RemoveTableItemButton
+                key={`remove-btn-${file.id}`}
+                onRemove={() => removeProjectReferenceFile(projectId, file.id) }
+                onRemoveSuccess={() => {
+                  updateReferenceFileTable();
+                }}
+                tooltipText={i18n("ReferenceFile.removeTooltip")}
+                confirmText={i18n("ReferenceFile.confirmText", file.label, projectName)}
+              />
+              :
+              null
+          ]
+        // Return download and remove buttons spaced
+        return (<Space size="small">{actionButtons}</Space>);
+      },
+    },
+  ];
+
+  // On first load of page call method to get the reference files for the project
+  React.useEffect(() => {
+    updateReferenceFileTable();
+  }, []);
+
+  // Get the reference files for the project
+  function updateReferenceFileTable(){
+    getProjectReferenceFiles(projectId).then(({files}) => {
+      setProjectReferenceFiles(files);
+      setLoading(false);
+    }).catch()
+  }
+
+  // Options for the Ant Design upload component
+  const referenceFileUploadOptions = {
+    multiple: true,
+    showUploadList: false,
+    action: setBaseUrl(`ajax/referenceFiles/project/${projectId}/new`),
+    onChange(info) {
+      const { status } = info.file;
+      if (status === "done") {
+        notification.success({message: `${i18n("ReferenceFile.uploadFileSuccess", info.file.name, projectName)}`});
+        updateReferenceFileTable();
+      } else if (status === "error") {
+        notification.error({message: info.file.response.error});
+      }
+    },
+  };
+
+  /*
+   * Returns the upload section if a user is allowed to manage the project.
+   * Supports drag and drop as well as click to upload
+   */
+  function displayUploadButton() {
+     if (canManage)
+      return (
+        <Dragger {...referenceFileUploadOptions} style={{marginBottom: SPACE_XS}}>
```

Comment:
Can you make our own component out of this?  I am surprised the ant design does not seem to have one since is has the classes set up for it.

Reply:
Yup created in 4e0f1cd19cfb95ba3af35a86f7e3a2271d907412